src/pubsub/ua_pubsub_reader.c

```diff
@@ -373,7 +373,14 @@ checkReaderIdentifier(UA_Server *server, UA_NetworkMessage *msg,
             }
         }
         if(msg->payloadHeaderEnabled) {
-            if(reader->config.dataSetWriterId != *msg->payloadHeader.dataSetPayloadHeader.dataSetWriterIds) {
+            UA_Byte anzDataSets = msg->payloadHeader.dataSetPayloadHeader.count;
```

Comment:
Is "anz" for the German "Anzahl"?
Keep to English also for variable names.

Reply:
changed to 'totalDataSets'